#! /usr/bin/env python
#Autogenerated V4L Dynamic Control

PACKAGE='ball_detection'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("low_h_r", int_t, 0, "H1 for red (low)", 0, 0, 180)
gen.add("high_h_r", int_t, 0, "H1 for red (high)", 6, 0, 180)
gen.add("low_h2_r", int_t, 0, "H2 for red (low)", 167, 0, 180)
gen.add("high_h2_r", int_t, 0, "H2 for red (high)", 180, 0, 180)
gen.add("low_s_r", int_t, 0, "S for red (low)", 90, 0, 255)
gen.add("high_s_r", int_t, 0, "S for red (high)", 255, 0, 255)
gen.add("low_v_r", int_t, 0, "V for red (low)", 102, 0, 255)
gen.add("high_v_r", int_t, 0, "V for red (high)", 255, 0, 255)

gen.add("low_h_b", int_t, 0, "H1 for blue (low)", 91, 0, 180)
gen.add("high_h_b", int_t, 0, "H1 for blue (high)", 119, 0, 180)
gen.add("low_s_b", int_t, 0, "S for blue (low)", 227, 0, 255)
gen.add("high_s_b", int_t, 0, "S for blue (high)", 255, 0, 255)
gen.add("low_v_b", int_t, 0, "V for blue (low)", 47, 0, 255)
gen.add("high_v_b", int_t, 0, "V for blue (high)", 255, 0, 255)

gen.add("a", double_t, 0, "Range calibration parameter a in y= ax+b", 1.0, 0.0, 2.0)
gen.add("b", double_t, 0, "Range calibration parameter b in y= ax+b", 0.0, -100.0, 100.0)

exit(gen.generate(PACKAGE, "ball_detection", "BallDetection"))
